CondeNast/launch-vehicle-fbm
# Add an option to allow the SDK to process echo messages

## Why are we doing this?

Two enable two bots one page for a new conversation logger, you need the ability to make a bot that can listen to echo messages. Otherwise, the observer bot can only observe humans.

## Did you document your work?

README has the new constructor option and some cleanup.

## How can someone test these changes?

Steps to manually verify the change:

Just unit tests. After this, I can open up the logger as an example bot. This requires an elaborate setup to manually verify.

## What possible risks or adverse effects are there?

* none

## What are the follow-up tasks?

* release a new version

## Are there any known issues?

* if you use npm v2, the peer dependencies will fail. Don't use npm 2. I thought about explicitly putting it in the package.json engines declaration but at this point, no one should run into this.

## Did the test coverage decrease?

stays the same. some improvements to the tests:
* `messenger` no longer leaks state
* document an assertion that double checks that your listener fired or not. If you have assertions in an event listener but the even never fires, you could get a passing tests that should have failed.
